src/onConsentChange.ts

```diff
@@ -53,8 +53,29 @@ const getConsentState: () => Promise<ComparedConsentState> = () =>
 		} else if (window.__tcfapi) {
 			// in RoW - https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md
 			window.__tcfapi('getTCData', 2, (tcfData, success) => {
-				if (success) {
-					resolve(compareState({ tcfv2: tcfData?.purpose.consents }));
+				if (success && window.__tcfapi) {
```

Comment:
can be refactored to use Promise.all and remove duplication of if (window.__tcfapi)

Reply:
Well done!